drivers/flash/nrf_qspi_nor.c

```diff
@@ -796,6 +796,7 @@ static int qspi_sfdp_read(const struct device *dev, off_t offset,
 out:
 	qspi_unlock(dev);
 	ANOMALY_122_UNINIT(dev);
+out_failed_init:
```

Comment:
This label should be located before the call to `ANOMALY_122_UNINIT()`, because of the counting semaphore used by `anomaly_122_init/uninit`.
But there is another problem here - the value returned by `ANOMALY_122_INIT()` should not be passed to `qspi_get_zephyr_ret_code()`, as it is already a Zephyr error code (BTW `res` should be of the `nrfx_err_t` type, not `int`, this would probably also help avoid confusion), so maybe it would be better to use something like:
```
	int rc = ANOMALY_122_INIT(dev);
	if (rc != 0) {
		LOG_DBG("ANOMALY_122_INIT: %x", rc);
		ANOMALY_122_UNINIT(dev);
		return;
	}

```

Reply:
Replacing the PR with https://github.com/zephyrproject-rtos/zephyr/pull/40017